vita3k/modules/SceGxm/SceGxm.cpp

```diff
@@ -3219,19 +3212,14 @@ EXPORT(int, sceGxmProgramGetOutputRegisterFormat, const SceGxmProgram *program,
     return 0;
 }
 
-EXPORT(Ptr<SceGxmProgramParameter>, sceGxmProgramGetParameter, Ptr<const SceGxmProgram> program, uint32_t index) {
+EXPORT(Ptr<SceGxmProgramParameter>, sceGxmProgramGetParameter, const SceGxmProgram *program, uint32_t index) {
     TRACY_FUNC(sceGxmProgramGetParameter, program, index);
 
-    const SceGxmProgram *program_ptr = program.get(emuenv.mem);
-    if (index >= program_ptr->parameter_count)
+    if (index >= program->parameter_count)
         return Ptr<SceGxmProgramParameter>(0);
 
-    Address parameter_addr = program.address();
-    parameter_addr += offsetof(SceGxmProgram, parameters_offset);
-    parameter_addr += program_ptr->parameters_offset;
-    parameter_addr += index * sizeof(SceGxmProgramParameter);
-
-    return Ptr<SceGxmProgramParameter>(parameter_addr);
+    auto parameters = program->program_parameters();
+    return Ptr<SceGxmProgramParameter>(&parameters[index], emuenv.mem);
```

Comment:
Same, I used this construction because Ptr(void*, mem) does not work with page tables (I should remove it or make it work properly with page tables although that's a bit annoying).

Reply:
Also my bad, I should have commented why I was doing this.

Reply:
Oh. I see. I think Ptr(void*, mem) should be fixed or removed. I'm going to do it.